addons/mail/models/mail_channel.py

```diff
@@ -238,12 +376,53 @@ def message_get_recipient_values(self, notif_message=None, recipient_ids=None):
             }
         return super(Channel, self).message_get_recipient_values(notif_message=notif_message, recipient_ids=recipient_ids)
 
+    def _extract_values(self, message_type, **kwargs):
+        """ This method is used to compute moderation status before the creation of a message. 
+            For this operation the message's author email address is required.
+            This address is returned with status for other computations.
+        """
+        moderation_status = 'accepted'
+        email = ''
+        if self.moderation and message_type in ['email', 'comment']:
+            author_id = kwargs.get('author_id')
+            if author_id and isinstance(author_id, pycompat.integer_types):
+                author = self.env['res.partner'].browse([author_id])
+                email = author.email
+            elif author_id:
+                email = author_id.email
+            elif self.env.user:
```

Comment:
As you should always have a user in the environment (being admin for the mail gateway) you will never fall in the last branch of your conditional tree. Checking email_from from kwargs should be done before using env.user I think.

Reply:
Need to check if it is fine now!